### PR TITLE
CI: add macos-15-large

### PR DESCRIPTION
**Pull request by AkihiroSuda**

### CODE REVIEW

**Comment by AkihiroSuda** [comment]
https://github.com/lima-vm/socket_vmnet/actions/runs/11668029741/job/32487531768?pr=63

```
▶ Run limactl shell vm1 ip -4 -json addr show dev lima0 | jq -r .[0].addr_info[0].local | tee /tmp/vm1_iP
192.168.105.3
```

```
▶
Run iperf3 -c "$(cat /tmp/vm1_ip)"
  
iperf3: error - unable to connect to server - server may have stopped running or use a different port, firewall issue, etc.: No route to host
```

**Review by jandubois — APPROVED** [review]
Thanks, LGTM

**Comment by AkihiroSuda** [comment]
https://github.com/lima-vm/socket_vmnet/actions/runs/11670466432/job/32494724116?pr=63

```
+ netstat -rn
Routing tables

Internet:
Destination        Gateway            Flags               Netif Expire
default            192.168.64.1       UGScg                 en0       
default            link#11            UCSIg           bridge100      !
127                127.0.0.1          UCS                   lo0       
127.0.0.1          127.0.0.1          UH                    lo0       
169.254            link#5             UCS                   en0      !
192.168.64         link#5             UCS                   en0      !
192.168.64.1/32    link#5             UCS                   en0      !
192.168.64.1       16:9d:99:c7:98:64  UHLWIir               en0   1144
192.168.64.30/32   link#5             UCS                   en0      !
192.168.105        link#11            UC              bridge100      !
192.168.105.2      link#11            UHLWIi          bridge100      !
192.168.105.3      52.55.55.e7.4b.ea  UHLWIi          bridge100   1186
[...]

++ cat /tmp/vm1_ip
+ ping -c 3 192.168.105.3
PING 192.168.105.3 (192.168.105.3): 56 data bytes
64 bytes from 192.168.105.3: icmp_seq=0 ttl=64 time=1.498 ms
64 bytes from 192.168.105.3: icmp_seq=1 ttl=64 time=1.661 ms
64 bytes from 192.168.105.3: icmp_seq=2 ttl=64 time=3.294 ms

--- 192.168.105.3 ping statistics ---
3 packets transmitted, 3 packets received, 0.0% packet loss
round-trip min/avg/max/stddev = 1.498/2.151/3.294/0.811 ms
++ cat /tmp/vm1_ip
+ iperf3 -c 192.168.105.3
iperf3: error - unable to connect to server - server may have stopped running or use a different port, firewall issue, etc.: No route to host
```

The routing table seems just same as macos-13.

"No route to host" doesn't seem to occur with `ping` 🤔

**Comment by nirs** [comment]
I would split the changes to:
- remove macos-12 to ubreak the build
- add macos-14 since it works now
- add macos-15 later, probably need more work

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -84,13 +84,24 @@ jobs:
           limactl shell vm1 ip a
       - name: "Lima: vm1: set up iperf3"
         run: |
-          limactl shell vm1 sudo DEBIAN_FRONTEND=noninteractive apt-get install -y iperf3
+          # net-tools for netstat
+          limactl shell vm1 sudo DEBIAN_FRONTEND=noninteractive apt-get install -y iperf3 net-tools
           limactl shell vm1 systemd-run --user --unit=iperf3 iperf3 -s
```
Comment (nirs) [inline .github/workflows/test.yml:89]: why not use the iperf3 service?

    limactl shell sudo systemctl enable iperf3 --now

**Comment by nirs** [comment]
Should work after #83 is merged.

File: .github/workflows/test.yml [hunk 2]
```diff
@@ -11,18 +11,24 @@ permissions:
...
+      - uses: actions/checkout@eef61447b9ff4aafe5dcd4e0bbf5d482be7e7871 # v4.2.1
+        with:
+          fetch-depth: 1
+      - name: Build
+        run: make
+  integration:
+    name: Integration tests
+    # Only a single version of macOS is used for running integration tests, due to the flakiness of the CI.
+    runs-on: macos-15-large  # Intel (supports nested virtualization)
     timeout-minutes: 40
```
Comment (nirs) [inline .github/workflows/test.yml:32]: The build takes 9 minutes now (with #84) - so we can use much smaller timeout now, maybe 15 minutes?

File: .github/workflows/test.yml [hunk 3]
```diff
@@ -11,18 +11,24 @@ permissions:
...
     strategy:
       fail-fast: false
       matrix:
-        # macos-13-large is used as macos-13 seems too flaky.
-        # macos-14 (ARM) and later cannot be used for the most part of the job
-        # due to the lack of the support for nested virt.
-        #
-        # TODO: add macos-15-large https://github.com/lima-vm/socket_vmnet/pull/63
-        platform: [macos-13-large, macos-14-large]
+        platform: [macos-13, macos-14, macos-15]
```
Comment (nirs) [inline .github/workflows/test.yml:19]: This build only on aarch64, previously we built only on x86_64. We don't wan to test build on both?

File: .github/workflows/test.yml [hunk 4]
```diff
@@ -11,18 +11,24 @@ permissions:
...
+          fetch-depth: 1
+      - name: Build
+        run: make
+  integration:
+    name: Integration tests
+    # Only a single version of macOS is used for running integration tests, due to the flakiness of the CI.
+    runs-on: macos-15-large  # Intel (supports nested virtualization)
     timeout-minutes: 40
     steps:
       - uses: actions/checkout@eef61447b9ff4aafe5dcd4e0bbf5d482be7e7871 # v4.2.1
```
Comment (nirs) [inline .github/workflows/test.yml:34]: This compiles both for x86_64 and aarch64, but we test only on x86_64 - any reason to build twice when we test the build in the build job?

File: .github/workflows/test.yml [hunk 5]
```diff
@@ -11,18 +11,24 @@ permissions:
...
+    timeout-minutes: 10
+    steps:
+      - uses: actions/checkout@eef61447b9ff4aafe5dcd4e0bbf5d482be7e7871 # v4.2.1
+        with:
+          fetch-depth: 1
+      - name: Build
+        run: make
+  integration:
+    name: Integration tests
+    # Only a single version of macOS is used for running integration tests, due to the flakiness of the CI.
```
Comment (nirs) [inline .github/workflows/test.yml:30]: Good idea but better to add macos-15 and change the build in separate commits.

Comment (AkihiroSuda) [reply .github/workflows/test.yml:30]: Why?

File: .github/workflows/test.yml (hunk 3 above)
Comment (AkihiroSuda) [reply .github/workflows/test.yml:19]: Intel is covered in the integration test

File: .github/workflows/test.yml (hunk 4 above)
Comment (nirs) [reply .github/workflows/test.yml:34]: OK, makes sense, this is also very quick.

**Review by nirs — APPROVED** [review]
Lets merge, we can always add and remove platforms.

File: .github/workflows/test.yml (hunk 3 above)
Comment (nirs) [reply .github/workflows/test.yml:19]: This will be easier to maintain if we use:

```yaml
platform:
- macos-13
- macos-14
- macos-15
```

**Pull request merged**